deps/src/main/java/com/microsoft/azure/sdk/iot/deps/serializer/FileUploadCompletionNotification.java

```diff
@@ -0,0 +1,195 @@
+
+/**
+ * The request payload to send to IoT Hub to notify it when a file upload is completed, whether successful or not.
+ * Must set {@link #setSuccess(Boolean)} and {@link #setCorrelationId(String)}, but all other fields are optional.
+ */
+public class FileUploadCompletionNotification
+{
+    private static final String CORRELATION_ID_TAG = "correlationId";
+    @Expose(serialize = true, deserialize = true)
+    @SerializedName(CORRELATION_ID_TAG)
+    private String correlationId = null;
+
+    private static final String IS_SUCCESS_TAG = "isSuccess";
+    @Expose(serialize = true, deserialize = true)
+    @SerializedName(IS_SUCCESS_TAG)
+    private Boolean isSuccess = null;
+
+    private static final String STATUS_CODE_TAG = "statusCode";
+    @Expose(serialize = true, deserialize = true)
+    @SerializedName(STATUS_CODE_TAG)
+    private Integer statusCode = null;
+
+    private static final String STATUS_DESCRIPTION_TAG = "statusDescription";
+    @Expose(serialize = true, deserialize = true)
+    @SerializedName(STATUS_DESCRIPTION_TAG)
+    private String statusDescription = null;
+
+    /**
+     * Create an instance of the FileUploadCompletionNotification for a single file upload operation using Azure Storage.
+     *
+     * @param correlationId the correlationId that correlates this FileUploadCompletionNotification to the earlier request to get the SAS URI
+     *                      for this upload from IoT Hub. This field is mandatory. Must equal {@link FileUploadSasUriResponse#getCorrelationId()}.
+     * @param isSuccess whether the file was uploaded successfully. This field is mandatory.
+     * @throws IllegalArgumentException if one of the parameters is null, empty, or not valid.
+     */
+    public FileUploadCompletionNotification(String correlationId, Boolean isSuccess)
+            throws IllegalArgumentException
+    {
+        setCorrelationId(correlationId);
+        updateStatus(isSuccess, statusCode, statusDescription);
+    }
+
+    /**
+     * Create an instance of the FileUploadCompletionNotification for a single file upload operation using Azure Storage.
+     *
+     * @param correlationId the correlationId that correlates this FileUploadCompletionNotification to the earlier request to get the SAS URI
+     *                      for this upload from IoT Hub. This field is mandatory. Must equal {@link FileUploadSasUriResponse#getCorrelationId()}.
+     * @param isSuccess whether the file was uploaded successfully. This field is mandatory.
+     * @param statusCode is the status for the upload of the file to storage.
+     * @param statusDescription is the description of the status code.
+     * @throws IllegalArgumentException if one of the parameters is null, empty, or not valid.
+     */
+    public FileUploadCompletionNotification(String correlationId, Boolean isSuccess, Integer statusCode, String statusDescription)
+            throws IllegalArgumentException
```

Comment:
is this 'throws' needed since the method doesn't throw anything?

Reply:
Good catch, there is nothing thrown here anymore so I'll update the method signature and the javadocs